File: app/src/main/java/com/example/pam_app/db/WallyDatabase.java
package com.example.pam_app.db;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = {BucketEntity.class, BucketEntryEntity.class, IncomeEntity.class}, version = 9, exportSchema = false)
public abstract class WallyDatabase extends RoomDatabase {

    private static final String NAME = "wally_db";

    public abstract BucketDao bucketDao();
    public abstract IncomeDao incomeDao();

    public static synchronized WallyDatabase getInstance(final Context context) {
        return Room.databaseBuilder(context.getApplicationContext(), WallyDatabase.class, NAME)
                            .fallbackToDestructiveMigration().build();
    }
}

File: app/src/main/java/com/example/pam_app/db/WallyDatabase.kt
package com.example.pam_app.db

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

@Database(entities = [BucketEntity::class, BucketEntryEntity::class, IncomeEntity::class], version = 11, exportSchema = false)
abstract class WallyDatabase : RoomDatabase() {
    abstract fun bucketDao(): BucketDao?
    abstract fun incomeDao(): IncomeDao?

    companion object {
        private const val NAME = "wally_db"

        @Synchronized
        fun getInstance(context: Context): WallyDatabase {
            return Room.databaseBuilder(context.applicationContext, WallyDatabase::class.java, NAME)
                    .fallbackToDestructiveMigration().build()
        }
    }
}